stacks-common/src/util/macros.rs

```diff
@@ -220,8 +220,10 @@ macro_rules! define_u8_enum {
         #[derive(Debug, Clone, Copy, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
         #[repr(u8)]
         $(#[$outer])*
+        /// A u8 Named Enum
         pub enum $Name {
             $(  $(#[$inner])*
+                /// $Variant
```

Comment:
This macro should support rustdocs in the macro applications, e.g.,

```rust
define_u8_enum!(
/// Enum representing the stackerdb message identifier: this is
///  the contract index in the signers contracts (i.e., X in signers-0-X)
MessageSlotID {
    /// DkgBegin message
    DkgBegin = 0,
    /// DkgPrivateBegin
    DkgPrivateBegin = 1,
    /// DkgEndBegin
    DkgEndBegin = 2,
    /// DkgEnd
```

Which is probably better to do than to add programmatic documentation.

Reply:
ah for some reason I could not get this to work but am almost 90 percent sure its cause I put it ABOVE define_u8_enum. Will do.